fe/fe-core/src/main/java/com/starrocks/analysis/ListPartitionDesc.java

```diff
@@ -0,0 +1,226 @@
+        for (MultiItemListPartitionDesc desc : this.multiListPartitionDescs) {
+            if (!multiListPartitionName.add(desc.getPartitionName())) {
+                throw new AnalysisException("Duplicated partition name: " + desc.getPartitionName());
+            }
+            desc.analyze(this.partitionColNames.size(), tableProperties);
+            allMultiValues.addAll(desc.getMultiValues());
+        }
+        this.analyzeDuplicateValues(this.partitionColNames.size(), allMultiValues);
+    }
+
+    private void analyzeSingleListPartition(Map<String, String> copiedProperties) throws AnalysisException {
+        List<String> allValues = Lists.newArrayList();
+        Set<String> singListPartitionName = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
+        for (SingleItemListPartitionDesc desc : this.singleListPartitionDescs) {
+            if (!singListPartitionName.add(desc.getPartitionName())) {
+                throw new AnalysisException("Duplicated partition name: " + desc.getPartitionName());
+            }
+            desc.analyze(this.partitionColNames.size(), copiedProperties);
+            allValues.addAll(desc.getValues());
+        }
+        this.analyzeDuplicateValues(allValues);
+    }
+
+    /**
+     * Check if duplicate values are found
+     * If the value of the member in the same position is equals, it is considered a duplicate
+     * @param partitionColSize the partition column size
+     * @param allMultiValues values from multi list partition
+     * @throws AnalysisException
+     */
+    private void analyzeDuplicateValues(int partitionColSize, List<List<String>> allMultiValues) throws AnalysisException {
+        List<List<String>> tempMultiValues = new ArrayList<>(allMultiValues.size());
+        for (List<String> values : allMultiValues) {
+            for (List<String> tempValues : tempMultiValues){
+                int duplicatedSize = 0;
+                for (int i = 0; i < values.size(); i++) {
+                    if(values.get(i).equals(tempValues.get(i))){
+                        duplicatedSize++;
+                    }
+                }
+                if (duplicatedSize == partitionColSize) {
+                    throw new AnalysisException("Duplicate values " +
+                            "(" + String.join(",", values) + ") not allow");
+                }
+            }
+            tempMultiValues.add(values);
+        }
+    }
+
+    /**
+     * Check if duplicate values are found
+     * Use hashSet to check duplicate value
+     * @param allValues values from single list partition
+     * @throws AnalysisException
+     */
+    private void analyzeDuplicateValues(List<String> allValues) throws AnalysisException {
+        Set<String> hashSet = new HashSet<>();
+        for (String value : allValues) {
+            if (!hashSet.add(value)) {
+                throw new AnalysisException("Duplicated value" + value);
+            }
+        }
+    }
```

Comment:
Use `LiteralExpr` to check.
for example, partition column is DATETIME type, and partition value "2022-04-01" and "2022-04-01 00:00:00" should be same.